OffscreenCanvas WebGL does not produce console log messages

kkinnunen-apple opened the pull request:
#### c8113478349e7c301c177862a4a70e75f1bd9fc4
<pre>
OffscreenCanvas WebGL does not produce console log messages
<a href="https://bugs.webkit.org/show_bug.cgi?id=261400">https://bugs.webkit.org/show_bug.cgi?id=261400</a>
rdar://115275708

Reviewed by Dan Glastonbury.

Use ScriptExecutionContext of the CanvasBase to add the console message.
This works for HTMLElement as well as OffscreenCanvas.

Do not reset the error counter on context restore. The counter is logical
to be for the context object, not for the underlying OpenGL context.

* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
(WebCore::WebGLRenderingContextBase::WebGLRenderingContextBase):
(WebCore::WebGLRenderingContextBase::initializeContextState):
(WebCore::WebGLRenderingContextBase::shouldPrintToConsole const):
(WebCore::WebGLRenderingContextBase::printToConsole):
* Source/WebCore/html/canvas/WebGLRenderingContextBase.h:

Canonical link: <a href="https://commits.webkit.org/267895@main">https://commits.webkit.org/267895@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6966101301dd2d2e563ded170299b71e794a38a7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/17897 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/18223 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/18783 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/19725 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16744 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/18093 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/21516 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/18378 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/18752 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18113 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18376 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/15562 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/20593 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/15603 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/16315 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/22847 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/16619 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/16484 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20713 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17047 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/14440 "1 failures") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16148 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/16156 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4283 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/20507 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/16895 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

djg commented on LayoutTests/webgl/offscreen-webgl-errors-to-console.html:
```diff
@@ -0,0 +1,34 @@
+<body>
+<script src="../resources/js-test-pre.js"></script>
+<script>
+// Tests that using offscreen canvas WebGL, we log a warning to the console if the setting is enabled.
```
"For tests that use offscreen canvas WebGL, we [...]"

kkinnunen-apple replied:
Not a native english speaker, but that means different thing.

What your suggestion is saying: "For red shirts that are dirty we use washing machine", e.g. it states the universal fact of the engine, qualified to subset of tests redundantly. We use washing machine for all shirts, we log console messages for all webgl tests.

What mine intended to communicate is
"[This test in the lines below] Tests that using offscreen canvas WebGL, we log a warning to the console if the setting is enabled."

djg reviewed (approved):
:shipit:

kkinnunen-apple committed:
OffscreenCanvas WebGL does not produce console log messages
https://bugs.webkit.org/show_bug.cgi?id=261400
rdar://115275708

Reviewed by Dan Glastonbury.

Use ScriptExecutionContext of the CanvasBase to add the console message.
This works for HTMLElement as well as OffscreenCanvas.

Do not reset the error counter on context restore. The counter is logical
to be for the context object, not for the underlying OpenGL context.

* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
(WebCore::WebGLRenderingContextBase::WebGLRenderingContextBase):
(WebCore::WebGLRenderingContextBase::initializeContextState):
(WebCore::WebGLRenderingContextBase::shouldPrintToConsole const):
(WebCore::WebGLRenderingContextBase::printToConsole):
* Source/WebCore/html/canvas/WebGLRenderingContextBase.h:

Canonical link: https://commits.webkit.org/267895@main
